### PR TITLE
PoC @alias annotation

### PR DESCRIPTION
**Pull request by erlend-aasland**
TODO:

- tests
- more guards for special cases

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @eltoder

**Comment by erlend-aasland** [comment]
As discussed in the upstream issue, this PR confirms that the `@alias` notation is straight-forward to implement. There are some corner cases to get rid of, and of course tests, before we can propose this on the CPython repo.

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -1769,6 +1769,18 @@ def parser_body(
...
         }
+        if f.alias:
+            # Duplicate the PyMethodDef define, and replace the
+            # 'methoddef_name' field with 'methoddef_alias'. During generation,
+            # 'methoddef_alias' will be swapped with f.alias (from @alias).
+            alias_def = methoddef_define
+            _, cname = f.alias
+            if cname:
+                alias_def = alias_def.replace("methoddef_name",
+                                              "methoddef_alias")
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py:1780]: This feels somewhat hacky, but I guess no more hacky than all the other times we're calling `.replace()` on templates in `clinic.py` :)

**Review by AlexWaygood — COMMENTED** [review]
Seems reasonable!

File: Tools/clinic/clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Tools/clinic/clinic.py:1780]: Yeah, it is a bit hacky. We could add some `assert`s, to make sure the code breaks sooner than later if the prototype template changes.

**Comment by erlend-aasland** [comment]
I'll add some guards and tests and move this upstream.

**Comment by erlend-aasland** [comment]
> I'll add some guards and tests and move this upstream.

As noted on the upstream issue, we need to do some more refactoring before we can finally upstream this.
